src/util/RenameThread.cpp

```diff
@@ -0,0 +1,22 @@
+#include "util/RenameThread.hpp"
+
+#include "common/QLogging.hpp"
+
+#ifdef Q_OS_WIN
+#    include <Windows.h>
+#endif
+
+namespace chatterino::windows::detail {
+
+void renameThread(HANDLE hThread, const QString &threadName)
+{
+    auto hr = SetThreadDescription(hThread, threadName.toStdWString().c_str());
+    if (!SUCCEEDED(hr))
+    {
+        qCWarning(chatterinoCommon).nospace()
+            << "Failed to set thread description, hresult=0x"
+            << QString::number(hr, 16);
+    }
+}
```

Comment:
needs to be if-deffed - doesn't build on non-windows platforms

and you wanted to check if you need to do a win7 or something check too